doc/static-route/SONiC_static_route_hdl.md

```diff
@@ -0,0 +1,452 @@
+blackhole           =  string; List of boolean; true if the next-hop route is blackholed.                     
+```
+The nexthop-vrf if set, will allow to create a leaked route in the current VRF(vrf-name). The vrf-name, nexthop-vrf, prefix, ifname, distance and blackhole are all parameters required to configure static routes and route leaks in vtysh.
+If an interface is moved from one VRF to another and it exist in the STATIC_ROUTE table, the configured routes with that interface will become inactive. These routes are stored in vtysh "running config" and STATIC_ROUTE table. It is up to the user to remove stale/inactive routes. They will become active if/when the interface is bound again to the orignal VRF.
+
+In this table each entry based on the index in the lists of  nexthop, ifname, distance, nexthop-vrf and blackhole  is a set. Together based on the index they specify one next-hop entry as there can be multiple next-hops for a prefix. Empty string will be used to complete the set if required. Prefix and nexthop in a set is expected to have same address family.
+
+In the example -1 table below (0.0.0.0, Ethernet0, 10, default, false) , (2.2.2.1, "", 0,"", false), (0.0.0.0, Ethernet12, 30, Vrf-GREEN, false) are the 3 sets defined for 3 configured next-hops for prefix 10.11.1.0/24 in Vrf-RED.
+In example -2  the blackhole is true for second index and  only distance is relevant in blackholed route. Due to blackhole entry (with metric 40) all packets matching 10.12.11.0/24 will be discarded if other specified  nexthop is inactive.
+
+```JSON
+Example -1 :--
+
+key                 = STATIC_ROUTE|Vrf-RED|10.11.1.0/24;
+vrf-name            = Vrf-RED
+prefix              = 10.11.1.0/24
+nexthop             = 0.0.0.0, 2.2.2.1, 0.0.0.0
+ifname              = Ethernet0,"", Ethernet12
+distance            = 10,0,30
+nexthop-vrf         = default,"",Vrf-GREEN
+blackhole           = false, false, false
+
+EXAMPLE 2:-
+key                 = STATIC_ROUTE|default|10.12.11.0/24;
+vrf-name            = default
+prefix              = 10.12.11.0/24
+nexthop             = 2.2.2.3,""
+ifname              = "",""
+distance            = 10,40
+nexthop-vrf         = "",""
+blackhole           = false, true
+
+
+```
+
+
+Note: This model is proposed in line with ROUTE_TABLE model in application DB used for route updates and "vrf hld" description of STATIC_ROUTE table (which is currently not in use). Some default values like 0.0.0.0 and empty strings are used to complete a set. See the CLI example later to see how the table is filled up step-by-step.
+
+
+### 3.2.2 APP DB
+A new table "UNRESOLVE_NEIGH_TABLE" in aplication DB will be used to write unresolved next-hop for staic routes. This table will be written by router/neighbor module of orchagent and consumed by nbrmgrd process. Once the next-hop resolution is done by sending netlink message or static route is removed, entry is removed from the "UNRESOLVE_NEIGH_TABLE". These changes are coming from Broadcom.

```

Comment:
You could use the recently implemented APP_NEIGH_RESOLVE_TABLE_NAME for this purpose. Please check.

Reply:
We would need the change present in the PR https://github.com/Azure/sonic-swss/pull/1704, we can fix the text accordingly.